lib/WebLedgerClient.js

```diff
@@ -0,0 +1,288 @@
+    if(!this.ledgerAgent) {
+      await this.getAgent();
+    }
+    const baseURL = this.ledgerAgent.service[service];
+    const serviceApi = create({baseURL});
+    const response = await serviceApi.post('/', document, {
+      httpsAgent: this.httpsAgent, headers: requestHeaders
+    });
+    if(response.problem) {
+      const error = new WebLedgerClientError(
+        'Error sending document.', 'NetworkError',
+        {baseURL, service, document});
+      if(response.problem === 'CLIENT_ERROR') {
+        error.details = {
+          baseURL, document, service, error: response.data,
+          status: response.status
+        };
+      } else {
+        error.details = {
+          baseURL, document, service, error: response.originalError
+        };
+      }
+      throw error;
+    }
+    return response.data;
+  }
+
+  /**
+   * Sends an operation to a Web Ledger node.
+   *
+   * @param {object} operation - the operation to send.
+   * @param {object} [headers={}] - the headers to pass through with request.
+   *
+   * @returns {Promise<object>} response object
+   */
+  async sendOperation({operation, headers = {}}) {
+    const service = 'ledgerOperationService';
+    return this.sendDocument({document: operation, service, headers});
+  }
+
+  /**
+   * Resolves with a list of ledger agents.
+   *
+   * @param {object} [headers={}] - the headers to pass through with request.
+   *
+   * @returns {Promise<Array<object>>}
+   */
+  async getAgents({headers = {}} = {}) {
+    const baseURL = `https://${this.hostname}`;
+    const requestHeaders = Object.assign(
+      {}, constants.DEFAULT_HEADERS, headers);
+    const agentsApi = create({baseURL});
+    const response = await agentsApi.get('/ledger-agents', {}, {
+      headers: requestHeaders, httpsAgent: this.httpsAgent
+    });
+    if(response.problem) {
+      if(response.status === 404) {
+        throw new WebLedgerClientError(
+          'Record not found.', 'NotFoundError',
+          {baseURL, status: response.status});
+      }
+      const error = new WebLedgerClientError(
+        'Error retrieving record.', 'NetworkError');
+      if(response.problem === 'CLIENT_ERROR') {
+        error.details = {
+          baseURL, error: response.data, status: response.status
+        };
+      } else {
+        error.details = {
+          baseURL, error: response.originalError, status: response.status
+        };
+      }
+      throw error;
+    }
+    return response.data.ledgerAgent;
+  }
+
+  /**
+   * Resolves with a single ledger agent for given options. If "id" is not
+   * specified the first ledger agent will be resolved.
+   *
+   * @param {string} [id] - the ledger agent ID.
+   *
+   * @returns {Promise<object>}
+   */
+  async getAgent({id} = {}) {
+    const agents = await this.getAgents();
+    let ledgerAgent;
+    if(!id) {
+      [ledgerAgent] = agents;
+    } else {
+      [ledgerAgent] = agents.filter(agent => agent.id === id);
```

Comment:
Untested, but I think this could be `find` since only interested in the first match.
```suggestion
      ledgerAgent = agents.find(agent => agent.id === id);
```